llvm-general/src/LLVM/General/OrcJIT.hs

```diff
@@ -12,7 +12,12 @@ module LLVM.General.OrcJIT (
     mangleSymbol,
     withIRCompileLayer,
     withModuleSet,
-    withObjectLinkingLayer
+    withObjectLinkingLayer,
+    addModuleSet,
+    createObjectLinkingLayer,
```

Comment:
> Leaving the problem up to the user of llvm-general, however, should be the option of last resort.

Just to make sure, I make my intentions clear: I am in no way arguing for removing the `with` abstraction. `with` is  great, it is simple, easy to understand and easy to use.

What I am arguing for is that instead of trying to cover 100% of the cases by some abstraction that will probably be significantly more complex than `with`, we use `with` to cover 95% of the cases and give users the possibility to build there own abstractions/integrate with existing abstractions.